### PR TITLE
[topgen] Use OrderedDict for top, xbar objects

### PR DESCRIPTION
**Pull request by eunchan**
This is related to PR #1552.

`dict` < python3.7 does not keep the order of the key. It results the
random changes of the generated hjson file. With OrderedDict (introduced
in Python2.7), the generated hjson files are remains in order.

### CODE REVIEW

**Comment by eunchan** [comment]
Hum... whatever I try, still the order of output key is not kept. Only way is to set `sort_keys` True when dump the hjson. I am not sure whether this is intrinsic of `hjson.dump` function or not.

File: util/topgen/lib.py [hunk 1]
```diff
@@ -160,7 +165,7 @@ def get_pad_list(padstr):
     width = first - last + 1
 
     for p in range(first, last + 1):
-        pads.append({"name": pad, "index": p})
+        pads.append(OrderedDict({"name": pad, "index": p}))
```
Comment (eunchan) [inline util/topgen/lib.py]: @imphil This still cause the generated hjson has `index` field first and `name` later... really weird to me..

**Comment by eunchan** [comment]
So... I think the best way to not randomly change the generated hjson files is to sort the key when writing to the file, WDYT @imphil @asb ?

File: util/topgen/lib.py (hunk 1 above)
Comment (imphil) [reply util/topgen/lib.py]: You create a normal dict with `{` in your ordered dict -- so by the time your data/value pairs reach the `OrderedDict` constructor, they are already unordered. 

Use this instead for a Python 3.5-compatible solution:

```suggestion
        pads.append(OrderedDict([("name", pad), ("index", p)]))
```

The same is true for all the other cases where you construct an `OrderedDict`.

**Comment by eunchan** [comment]
Uh-oh... Now I see the problem! Thanks Philipp!

**Comment by eunchan** [comment]
Anyone willing to give the feedback? :)

File: util/topgen/intermodule.py [hunk 2]
```diff
@@ -51,7 +52,7 @@ def elab_intermodule(topcfg):
     list_of_intersignals = []
 
     if "inter_signal" not in topcfg:
-        topcfg["inter_signal"] = {}
+        topcfg["inter_signal"] = OrderedDict({})
```
Comment (imphil) [inline util/topgen/intermodule.py]: You can remove the curly braces here and in similar cases in this patch.

Comment (eunchan) [reply util/topgen/intermodule.py]: Thanks! revised.

**Pull request merged**